File: WeekendWarriors/Questionnaires/Questionnaires/QLS/AST/Widgets/CheckBox.cs
using Questionnaires.UI.Widgets;
using Questionnaires.QL.AST.Types;

namespace Questionnaires.QLS.AST.Widgets
{
    public class CheckBox : Widget
    {
        public override QuestionWidget CreateWidget(BooleanType type)
        {
            return new CheckBoxWidget();
        }
    }
}

File: AlternativeQL/src/org/qls/ast/widget/Checkbox.java
package org.qls.ast.widget;

import org.ql.ast.type.BooleanType;

public class Checkbox extends Widget {
    public boolean isCompatibleWith(BooleanType booleanType) {
        return true;
    }

    @Override
    public <T, C> T accept(WidgetVisitor<T, C> visitor, C context) {
        return visitor.visitCheckbox(this, context);
    }
}
